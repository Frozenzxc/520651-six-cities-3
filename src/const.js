const OfferType = {
  ALL: `cities`,
  FAVORITES: `favorites`,
  NEARBY: `near`,
};

const MAX_NEARBY_OFFERS = 3;

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const ReviewPostingStatus = {
  POSTED: `POSTED`,
  ERROR: `ERROR`,
};

const AppRoute = {
  FAVORITES: `/favorites`,
  LOGIN: `/login`,
  PROPERTY: `/offer/:id`,
  ROOT: `/`,
};

export {MAX_NEARBY_OFFERS, AuthorizationStatus, AppRoute, ReviewPostingStatus, OfferType};
