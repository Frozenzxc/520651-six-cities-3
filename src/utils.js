export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const parseOffer = (data) => {
  let offer = {};
  offer.bedrooms = data[`bedrooms`];
  offer.city = data[`city`];
  offer.description = data[`description`];
  offer.goods = data[`goods`];
  offer.host = {};
  offer.host.avatar = data[`host`][`avatar_url`];
  offer.host.id = data[`host`][`id`];
  offer.host.isPro = data[`host`][`is_pro`];
  offer.host.name = data[`host`][`name`];
  offer.id = data[`id`];
  offer.images = data[`images`];
  offer.isFavorite = data[`is_favorite`];
  offer.isPremium = data[`is_premium`];
  offer.location = data[`location`];
  offer.maxAdults = data[`max_adults`];
  offer.previewImage = data[`preview_image`];
  offer.price = data[`price`];
  offer.rating = data[`rating`];
  offer.title = data[`title`];
  offer.type = data[`type`];

  return offer;
};

export const parseReview = (data) => {
  let review = {};
  review.comment = data[`comment`];
  review.date = data[`date`];
  review.id = data[`id`];
  review.rating = data[`rating`];
  review.user = {};
  review.user.avatar = data[`user`][`avatar_url`];
  review.user.id = data[`user`][`id`];
  review.user.isPro = data[`user`][`is_pro`];
  review.user.name = data[`user`][`name`];

  return review;
};

export const getCitiesList = (offers) => {
  return [...new Set(offers.map((offer) => offer.city.name))];
};

export const noop = () => {};
