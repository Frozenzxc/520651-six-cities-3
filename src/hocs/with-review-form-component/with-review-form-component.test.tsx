import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";
import {ReviewPostingStatus} from "../../const";
import withReviewFormComponent from "./with-review-form-component";
import {noop} from "../../utils";

const mockStore = configureStore([]);

const MockComponent = () => {

  return (
    <div>
    </div>
  );
};

const MockComponentWrapped = withReviewFormComponent(MockComponent);

it(`withReviewFormComponent is rendered correctly`, () => {

  const store = mockStore({
    [NameSpace.OFFERS]: {
      isFormBlocked: false,
      reviewPostingStatus: null,
    },
  });
  const tree = renderer.create((
    <Provider store={store}>
      <MockComponentWrapped
        blockForm={noop}
        id={4}
        isFormBlocked={false}
        postReview={noop}
        resetFormStatus={noop}
        reviewPostingStatus={ReviewPostingStatus.POSTED}
      >
      </MockComponentWrapped>
    </Provider>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
