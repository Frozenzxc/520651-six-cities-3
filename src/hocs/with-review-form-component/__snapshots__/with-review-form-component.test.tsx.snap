// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withReviewFormComponent is rendered correctly 1`] = `<div />`;
